tests/conftest.py

```diff
@@ -22,4 +22,12 @@
 # THE SOFTWARE.
 #
 
-import pytest  # noqa
+
+def pytest_addoption(parser):
+    parser.addoption(
+        "--adapter",
+        action="store",
+        default=None,
+        dest="adapter",
+        help="Pass an adapter for connection to an instrument needed for a test.",
```

Comment:
```suggestion
        help="Pass an adapter string for connection to an instrument needed for a test. Select the desired module to test with the -k option",
```
For clarity?

Reply:
You can either use -k or you can just invoke pytest with the module name `pytest test_some_instrument.py`.

Reply:
I also thought that simply invoking pytest with the module name unskips the test but it is not working for me:

```
(pymeasure) PS C:\Users\bastian\Documents\programming\pymeasure> pytest .\tests\instruments\dummy\test_dumm.py   
=================================================== test session starts ====================================================
platform win32 -- Python 3.11.0, pytest-7.2.0, pluggy-1.0.0
rootdir: C:\Users\bastian\Documents\programming\pymeasure
collected 0 items / 1 skipped                                                                                                

==================================================== 1 skipped in 0.00s ====================================================
```

where the `test_dummy.py` reads 

```python
import pytest

pytest.skip("Only works with connected hardware", allow_module_level=True)


def test_dummy():
    assert True
```

pytest is version 7.2.0

Reply:
`-k` option does not behave as i would expect either...

```
(pymeasure) PS C:\Users\bastian\Documents\programming\pymeasure> pytest -k test_dummy.py
=================================================== test session starts ====================================================
platform win32 -- Python 3.11.0, pytest-7.2.0, pluggy-1.0.0
rootdir: C:\Users\bastian\Documents\programming\pymeasure
collected 662 items / 1 error / 662 deselected / 10 skipped / 0 selected

========================================================== ERRORS ==========================================================
______________________________________ ERROR collecting tests/display/test_inputs.py _______________________________________
tests\display\test_inputs.py:28: in <module>
    from pymeasure.display.inputs import ScientificInput, BooleanInput, ListInput
..\..\..\miniconda3\envs\pymeasure\Lib\site-packages\pymeasure\display\__init__.py:30: in <module>
    from .manager import Manager
..\..\..\miniconda3\envs\pymeasure\Lib\site-packages\pymeasure\display\manager.py:29: in <module>
    from .Qt import QtCore
..\..\..\miniconda3\envs\pymeasure\Lib\site-packages\pymeasure\display\Qt.py:27: in <module>
    from pyqtgraph.Qt import QtGui, QtCore, QtWidgets, loadUiType  # noqa: F401
..\..\..\miniconda3\envs\pymeasure\Lib\site-packages\pyqtgraph\__init__.py:18: in <module>
    from .colors import palette
..\..\..\miniconda3\envs\pymeasure\Lib\site-packages\pyqtgraph\colors\palette.py:1: in <module>
    from ..Qt import QtGui
..\..\..\miniconda3\envs\pymeasure\Lib\site-packages\pyqtgraph\Qt\__init__.py:54: in <module>
    raise Exception("PyQtGraph requires one of PyQt5, PyQt6, PySide2 or PySide6; none of these packages could be imported.")
E   Exception: PyQtGraph requires one of PyQt5, PyQt6, PySide2 or PySide6; none of these packages could be imported.
===================================================== warnings summary =====================================================
..\..\..\miniconda3\envs\pymeasure\Lib\site-packages\pymeasure\adapters\telnet.py:25
  C:\Users\bastian\miniconda3\envs\pymeasure\Lib\site-packages\pymeasure\adapters\telnet.py:25: DeprecationWarning: 'telnetlib' is deprecated and slated for removal in Python 3.13
    import telnetlib

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================================= short test summary info ==================================================
ERROR tests/display/test_inputs.py - Exception: PyQtGraph requires one of PyQt5, PyQt6, PySide2 or PySide6; none of these packages could be imported.
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
================================= 10 skipped, 662 deselected, 1 warning, 1 error in 1.78s ==================================
(pymeasure) PS C:\Users\bastian\Documents\programming\pymeasure> 
```

Same with explicit filepaths or simply `pytest -k test_dummy`. I am probably doing something wrong here...

Reply:
> I also thought that simply invoking pytest with the module name unskips the test but it is not working for me:

Yeah, no, this is not how pytest works. `-k`, `-m` etc influence the _collection_ phase, i.e. which tests are considered for execution. Skipping comes after/separate from that.

Your second comment has an error in the collection phase, maybe try to fix that (you don't seem to have pyqt/pyside installed) to avoid "muddying the waters", maybe then we see better what is going wrong (also add `-v` to get more info on the tests)

Reply:
Thanks for pointing out these two phases that happen. Also did not realize that PySide/PyQt is not installed automatically via `pip install .`. 

Little off-topic: Any plans to make the Qt packages / GUI stuff optional, lets say to install it via `pip install pymeasure[gui]`? At the moment pyqtgraph is a requirement but is (I guess) not needed when not using the GUI. If this is something that you consider useful, I could have a look at it.